lm_eval/evaluator.py

```diff
@@ -112,7 +112,17 @@ def simple_evaluate(
             + "_rank" + str(lm.rank) + ".db",
         )
 
-    task_dict = lm_eval.tasks.get_task_dict(tasks, num_fewshot=num_fewshot)
+    task_dict = lm_eval.tasks.get_task_dict(tasks)
+    for task_name in task_dict.keys():
+        config = task_dict[task_name]._config
+        if num_fewshot is not None:
+            if config["num_fewshot"] > 0:
+                default_num_fewshot = config["num_fewshot"]
+                eval_logger.warning(
+                    f"Overwriting default num_fewshot of {task_name} from {default_num_fewshot} to {num_fewshot}"
+                )
+
+            task_dict[task_name]._config.__setitem__("num_fewshot", num_fewshot)
```

Comment:
We should 1. make Task._config -> `Task.config` since it’s not private
2. maybe worth defining an “overwrite” method or something like that to not use __setitem__ raw?

Reply:
Yeah it's probably better. Are you okay if we change `_config` to `config` in `lm_eval/api/task.py`?

Reply:
Also, just remembered that with `__setitem__` I can change
```
task_dict[task_name]._config.__setitem__("num_fewshot", num_fewshot)
```
to
```
task_dict[task_name]._config["num_fewshot"] = num_fewshot
```

Reply:
> Are you okay if we change _config to config in lm_eval/api/task.py?

Yeah totally! Or we can make it a property of the task class. No strong preference between the two